examples/message_passing/after.cpp

```diff
@@ -0,0 +1,59 @@
+// This example shows how to use caf::after
+
+#include "caf/all.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <iostream>
+#include <random>
+using std::cout;
+using std::endl;
+
+using namespace caf;
+
+// Sends random characters to buddy, and then waits for a letter back
+behavior noisy_buddy(event_based_actor* self, actor buddy) {
+  using namespace std::chrono_literals;
+  std::random_device rd;
+  std::minstd_rand gen{rd()};
+  const auto count = std::uniform_int_distribution<>(20, 100)(gen);
+  std::uniform_int_distribution<> dis(33, 126);
+  for (auto i = 0; i < count; i++) {
+    self->send(buddy, static_cast<char>(dis(gen)));
+  }
+  return {[self](std::string letter) {
+    cout << "Received a letter:" << endl << letter << endl;
+    self->quit();
+  }};
+}
+
+struct aggregator_state {
+  std::string letter;
+  caf::actor dest;
+};
+
+// Aggregates incoming characters and stores the sender, replies with the
+// reversed string when inactive for 100ms
+behavior aggregator(stateful_actor<aggregator_state>* self) {
+  using namespace std::chrono_literals;
+  return {
+    [=](char c) mutable {
+      self->state.dest = caf::actor_cast<caf::actor>(self->current_sender());
```

Comment:
This comes with many asterisks. For an example that introduces `after`, I think staying away from advanced tools like `actor_cast` would be a better choice.

In general, I think the setup with `aggregator` sending something back doesn't really add much and obfuscates what we actually want to show.

What would you think of streamlining this as follows:
- Rename aggregator to collector and simply let it print after:
  - The buffer reached some maximum (say, 40 characters).
  - A timeout occurs (and the buffer is not empty).
- Rename noisy_buddy to generator and have it simply terminate after it's done emitting messages (by returning `void`).

I think that would make the control flow and purpose of each piece much more clear. Also would allow spawning multiple generators and give users an easy path forward to extend/play with this example.

Reply:
I've modified the example, since it was more of a rewrite I'll resolve the previous conversations, but I did integrate the feedback where applicable.